doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,653 @@
+
+.. code-block:: python
+
+    def sample():
+        """Generate and return a random number.
+
+        The value is sampled from a continuos uniform distribution between
+        0 and 1.
+
+        Returns
+        -------
+        float
+            Random number generated.
+        """
+        return random.random()
+
+With more than one value:
+
+.. code-block:: python
+
+    def random_letters():
+        """Generate and return a sequence of random letters.
+
+        The length of the returned string is also random, and is also
+        returned.
+
+        Returns
+        -------
+        length : int
+            Length of the returned string.
+        letters : str
+            String of random letters.
+        """
+        length = random.randint(1, 10)
+        letters = ''.join(random.choice(string.ascii_lowercase)
+                          for i in range(length))
+        return length, letters
+
+If the method yields its value:
+
+.. code-block:: python
+
+    def sample_values():
+        """Generate an infinite sequence of random numbers.
+
+        The values are sampled from a continuos uniform distribution between
+        0 and 1.
+
+        Yields
+        ------
+        float
+            Random number generated.
+        """
+        while True:
+            yield random.random()
+
+
+Section 5: See also
+~~~~~~~~~~~~~~~~~~~
+
+This is an optional section, used to let users know about pandas functionality
+related to the one being documented. While optional, this section should exist
+in most cases, unless no related methods or functions can be found at all.
+
+An obvious example would be the `head()` and `tail()` methods. As `tail()` does
+the equivalent as `head()` but at the end of the `Series` or `DataFrame`
+instead of at the beginning, it is good to let the users know about it.
+
+To give an intuition on what can be considered related, here there are some
+examples:
+
+* `loc` and `iloc`, as they do the same, but in one case providing indices and
+  in the other positions
+* `max` and `min`, as they do the opposite
+* `iterrows`, `itertuples` and `iteritems`, as it is easy that a user looking
+  for the method to iterate over columns ends up in the method to iterate
+  over rows, and vice-versa
+* `fillna` and `dropna`, as both methods are used to handle missing values
+* `read_csv` and `to_csv`, as they are complementary
+* `merge` and `join`, as one is a generalization of the other
+* `astype` and `pandas.to_datetime`, as users may be reading the documentation
+  of `astype` to know how to cast as a date, and the way to do it is with
+  `pandas.to_datetime`
+* `where` is related to `numpy.where`, as its functionality is based on it
+
+When deciding what is related, you should mainly use your common sense and
+think about what can be useful for the users reading the documentation,
+especially the less experienced ones.
+
+When relating to other methods (mainly `numpy`), use the name of the module
```

Comment:
"When relating to other methods" -> was this intended to be "other libraries" or "other modules"?